### PR TITLE
Fix CTS not able to run to completion

### PR DESCRIPTION
**Pull request by rairatne**
* Disable operation causing CTS/VTS crashed
* Added validation check for zero and invalid dimensions
* Set all supported operations default to false
* Adds load model for all execute specs
* Add checks for ouput recived from remote infer
* Return INVALID_ARGUMENT for failures in setting PoolInfos

Tracked-On: OAM-111360

### CODE REVIEW

File: bsp_diff/cic_cloud/vendor/intel/external/project-celadon/nn-hal/0002-Fix-crashes-observed-during-VTS-execution.patch [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+                 }
+                 destTensor = plugin->getInputTensor(tensorIndex);
+@@ -613,8 +617,22 @@ static std::tuple<ErrorStatus, hidl_vec<V1_2::OutputShape>, Timing> executeSynch
+         void* destPtr = modelInfo->getBlobFromMemoryPoolOut(request, i, expectedLength);
+         if (preparedModel->mRemoteCheck && preparedModel->mDetectionClient &&
+             preparedModel->mDetectionClient->get_status()) {
+-            preparedModel->mDetectionClient->get_output_data(std::to_string(i), (uint8_t*)destPtr,
++            const auto& copied = preparedModel->mDetectionClient->get_output_data(std::to_string(i), (uint8_t*)destPtr,
+                                                              expectedLength);
++            if ( copied == 0 ) {
```
Comment (JeevakaPrabu) [inline bsp_diff/cic_cloud/vendor/intel/external/project-celadon/nn-hal/0002-Fix-crashes-observed-during-VTS-execution.patch:59]: no need for extra space after ( and before )

File: bsp_diff/cic_cloud/vendor/intel/external/project-celadon/nn-hal/0002-Fix-crashes-observed-during-VTS-execution.patch [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
++                ALOGE("Output not populated");
++                return {ErrorStatus::GENERAL_FAILURE, {}, kNoTiming};
++            } else if ( copied != expectedLength ) {
```
Comment (JeevakaPrabu) [inline bsp_diff/cic_cloud/vendor/intel/external/project-celadon/nn-hal/0002-Fix-crashes-observed-during-VTS-execution.patch:62]: same comment as previous one

File: bsp_diff/cic_cloud/vendor/intel/external/project-celadon/nn-hal/0002-Fix-crashes-observed-during-VTS-execution.patch [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
++                const auto& opDims = modelInfo->getOperand(outIndex).dimensions;
++                std::vector<size_t> dims;
++                for ( const auto& dim : opDims ) { dims.push_back(dim); }
```
Comment (JeevakaPrabu) [inline bsp_diff/cic_cloud/vendor/intel/external/project-celadon/nn-hal/0002-Fix-crashes-observed-during-VTS-execution.patch:65]: same comments as previous one

**Review by JeevakaPrabu — CHANGES_REQUESTED** [review]
Add Change-Id: for all newly added commits

**Review by akodanka — APPROVED** [review]
LGTM

**Review by JeevakaPrabu — APPROVED** [review]

**Pull request merged**